src/algebra/group_power.lean

```diff
@@ -150,47 +131,49 @@ by induction q with q ih; [refl, rw [nat.pow_succ, pow_succ, mul_comm, ih]]
 by induction m with m ih; [rw [add_monoid.zero_smul, zero_mul],
   rw [succ_smul', ih, nat.succ_mul]]
 
-/- commutative monoid -/
+/-!
+### Commutative (additive) monoid
+-/
 
 section comm_monoid
-variables [comm_monoid α] {β : Type*} [add_comm_monoid β]
+variables [comm_monoid M] [add_comm_monoid A]
 
-theorem mul_pow (a b : α) (n : ℕ) : (a * b)^n = a^n * b^n :=
+theorem mul_pow (a b : M) (n : ℕ) : (a * b)^n = a^n * b^n :=
 by induction n with n ih; [exact (mul_one _).symm,
   simp only [pow_succ, ih, mul_assoc, mul_left_comm]]
-theorem add_monoid.smul_add : ∀ (a b : β) (n : ℕ), n•(a + b) = n•a + n•b :=
-@mul_pow (multiplicative β) _
+theorem add_monoid.smul_add : ∀ (a b : A) (n : ℕ), n•(a + b) = n•a + n•b :=
+@mul_pow (multiplicative A) _
 
-instance pow.is_monoid_hom (n : ℕ) : is_monoid_hom ((^ n) : α → α) :=
+instance pow.is_monoid_hom (n : ℕ) : is_monoid_hom ((^ n) : M → M) :=
 { map_mul := λ _ _, mul_pow _ _ _, map_one := one_pow _ }
 
-instance add_monoid.smul.is_add_monoid_hom (n : ℕ) : is_add_monoid_hom (add_monoid.smul n : β → β) :=
+instance add_monoid.smul.is_add_monoid_hom (n : ℕ) : is_add_monoid_hom (add_monoid.smul n : A → A) :=
 { map_add := λ _ _, add_monoid.smul_add _ _ _, map_zero := add_monoid.smul_zero _ }
 
 end comm_monoid
 
 section group
-variables [group α] {β : Type*} [add_group β]
+variables [group G] [group H] [add_group A] [add_group B]
 
 section nat
 
-@[simp] theorem inv_pow (a : α) (n : ℕ) : (a⁻¹)^n = (a^n)⁻¹ :=
+@[simp] theorem inv_pow (a : G) (n : ℕ) : (a⁻¹)^n = (a^n)⁻¹ :=
```

Comment:
Of course `(a : G)` looks a bit weird. Everyone would expect `(g : G)`. But that's very local weirdness, where as the preceding names on the types were more of a global weirdness. Let's not change the names of the terms in this PR. There is already enough going on...